src/backend.rs

```diff
@@ -15,115 +15,78 @@ pub struct CairoBackend<'a> {
-        self.call_cairo(|c| {
-            c.set_source_rgba(
-                f64::from(color.rgb.0) / 255.0,
-                f64::from(color.rgb.1) / 255.0,
-                f64::from(color.rgb.2) / 255.0,
-                color.alpha,
-            )
-        })
+    fn set_color(&self, color: &BackendColor) {
+        self.context.set_source_rgba(
+            f64::from(color.rgb.0) / 255.0,
+            f64::from(color.rgb.1) / 255.0,
+            f64::from(color.rgb.2) / 255.0,
+            color.alpha,
+        );
     }
 
-    fn set_stroke_width(&self, width: u32) -> Result<(), DrawingErrorKind<CairoError>> {
-        self.call_cairo(|c| c.set_line_width(f64::from(width)))
+    fn set_stroke_width(&self, width: u32) {
+        self.context.set_line_width(f64::from(width));
     }
 
-    fn set_font<S: BackendTextStyle>(&self, font: &S) -> Result<(), DrawingErrorKind<CairoError>> {
-        let actual_size = font.size();
-        self.call_cairo(|c| {
-            match font.style() {
-                FontStyle::Normal => c.select_font_face(
-                    font.family().as_str(),
-                    FontSlant::Normal,
-                    FontWeight::Normal,
-                ),
-                FontStyle::Bold => {
-                    c.select_font_face(font.family().as_str(), FontSlant::Normal, FontWeight::Bold)
-                }
-                FontStyle::Oblique => c.select_font_face(
-                    font.family().as_str(),
-                    FontSlant::Oblique,
-                    FontWeight::Normal,
-                ),
-                FontStyle::Italic => c.select_font_face(
-                    font.family().as_str(),
-                    FontSlant::Italic,
-                    FontWeight::Normal,
-                ),
-            };
-            c.set_font_size(actual_size);
-        })
+    fn set_font<S: BackendTextStyle>(&self, font: &S) {
+        match font.style() {
+            FontStyle::Normal => self.context.select_font_face(
+                font.family().as_str(),
+                FontSlant::Normal,
+                FontWeight::Normal,
+            ),
+            FontStyle::Bold => self.context.select_font_face(
+                font.family().as_str(),
+                FontSlant::Normal,
+                FontWeight::Bold,
+            ),
+            FontStyle::Oblique => self.context.select_font_face(
+                font.family().as_str(),
+                FontSlant::Oblique,
+                FontWeight::Normal,
+            ),
+            FontStyle::Italic => self.context.select_font_face(
+                font.family().as_str(),
+                FontSlant::Italic,
+                FontWeight::Normal,
+            ),
+        };
+        self.context.set_font_size(font.size());
     }
 
-    pub fn new(context: &'a CairoContext, (w, h): (u32, u32)) -> Result<Self, CairoError> {
-        let ret = Self {
+    pub fn new(context: &'a CairoContext, (w, h): (u32, u32)) -> Self {
```

Comment:
This is considered a breaking change. So we probably want to keep this function returning a result until next release.